internal_event_handlers.go

```diff
@@ -89,6 +88,7 @@ type (
 		enableLoggingInReplay bool // flag to indicate if workflow should enable logging in replay mode
 
 		metricsScope tally.Scope
+		hostEnv      *hostEnvImpl
```

Comment:
Sounds like this is related to your earlier comment about using getHostEnvironment everywhere.

Creating a reference of host environment in the entry point and propagating that through all other workers and pollers makes testing a lot easier. So when we use a test workflow environment, we can create a task handler with the same host env instead of the global one for instance. Was almost impossible to get tests working without this.

This is also nice with design since it makes future refactoring easier and ensures all the objects bundled together use the same host environment They just all happen to use the global one since thats what the aggregated worker decides.